awscrt/_test.py

```diff
@@ -0,0 +1,157 @@
+"""
+Private utilities for testing
+"""
+import _awscrt
+from awscrt import NativeResource
+import gc
+import inspect
+import os
+import sys
+import time
+import types
+
+
+def native_memory_usage() -> int:
+    """
+    Returns number of bytes currently allocated by awscrt's native code.
+
+    `AWS_CRT_MEMORY_TRACING `environment variable must be set before module
+    is loaded, or 0 will always be returned. Legal values are:
+
+    *   `AWS_CRT_MEMORY_TRACING=0`: No tracing
+
+    *   `AWS_CRT_MEMORY_TRACING=1`: Only track allocation sizes and total allocated
+
+    *   `AWS_CRT_MEMORY_TRACING=2`: Capture callstacks for each allocation
+    """
+    return _awscrt.native_memory_usage()
+
+
+def dump_native_memory():
+    """
+    If there are outstanding allocations from awscrt's native code, dump them
+    to log, along with any information gathered based on the tracing level.
+
+    In order to see the dump, logging must initialized at `LogLevel.Trace`
+    and the `AWS_CRT_MEMORY_TRACING` environment variable must be non-zero
+    when module is loaded. Legal values are:
+
+    *   `AWS_CRT_MEMORY_TRACING=0`: No tracing
+
+    *   `AWS_CRT_MEMORY_TRACING=1`: Only track allocation sizes and total allocated
+
+    *   `AWS_CRT_MEMORY_TRACING=2`: Capture callstacks for each allocation
+    """
+    return _awscrt.native_memory_dump()
+
+
+def join_all_native_threads(*, timeout_sec: float = -1.0) -> bool:
```

Comment:
Hiding some of these new functions in a private `_test.py` file so that we're free to change them going forward. If we have a need to make them public we can move them to `common.py`